net/peer.go

```diff
@@ -28,3 +36,57 @@ func CreatePeer(addr string, tls bool) Peer {
 		tls:  tls,
 	}
 }
+
+// RemoteAddress returns the address of the peer by first taking the address
+// that gRPC returns. If that address is a reserved address, then it tries to
+// read the "X-REAL-IP" header content.
+// For example, a valid nging config could include
+//
+// ```
+// location / {
+//       grpc_pass grpc://127.0.0.1:9091;
+//       grpc_set_header X-Real-IP $remote_addr;
+// }
+// ```
+//
+func RemoteAddress(c context.Context) string {
+	peer, ok := peer.FromContext(c)
+	var str string = ""
+	if ok {
+		str = peer.Addr.String()
+	}
+	// https://en.wikipedia.org/wiki/Reserved_IP_addresses
+	reserved := []string{
+		"192.168",
+		"10.0.0",
+		"127.0.0",
+		"192.0.0",
+		"192.168",
+	}
+
+	lookAtHeader := func() bool {
+		if str == "" {
+			return true
+		}
+		for _, r := range reserved {
+			if strings.HasPrefix(str, r) {
+				return true
+			}
+		}
+		return false
+	}()
```

Comment:
```
var lookAtHeader bool
if str == "" {
   lookAtHeader = true
} else {
   	for _, r := range reserved {
			if strings.HasPrefix(str, r) {
				lookAtHeader = true
        break
		}
}

Reply:
IDK which one is cleaner.

Reply:
changed to your example - i dont like for loops in if/else but it's better than calling a function.. :D